### ISSUE TITLE
Bug: CapturingSlot is not working for nullable types

### ISSUE THREAD

**Issue by stengvac** [mockk/mockk#293]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer (found captureNullable function)
- [x] I checked to make sure that this issue has not already been filed

### Current Behavior
Hello I am not sure whether following behavior is by design or not, but why  
``
class CapturingSlot<T : Any>()
`` require not null type? I have found, that I can capture nullable types with fun ``captureNullable(list)`` but can't with ``CapturingSlot``. So when I change captured type I have to use ``captureNullable(list)`` instead of ``capture(slot)``? I did not found definite answer in doc/google. 

Example

```
@ExtendWith(MockKExtension::class)
class SomeTest {
    val strSlot = slot<String>()
    
    @Test
    fun `no answer found for null`() {
        val anotherServiceMocked = mockk<AnotherService>() {
            every { callMe(capture(strSlot)) } just Runs
        }
        
        Tested(anotherServiceMocked).testedFnc(null)
    }

    @Test
    fun `answer found for not null`() {
        val anotherServiceMocked = mockk<AnotherService>() {
            every { callMe(capture(strSlot)) } just Runs
        }

        Tested(anotherServiceMocked).testedFnc("")
    }
}

class Tested(
    private val anotherService: AnotherService
) {
    fun testedFnc(str: String?) = anotherService.callMe(str)
}

class AnotherService {
    fun callMe(str: String?) = Unit
}
```

### Expected behavior
``
class CapturingSlot<T : Any?>()
`` 
so ``val stringSlot = slot<String?>()`` is possible to create and use to capture values. And capture will behave same for for null and not null values

### SW
MockK version: 1.9.3
OS: Win 10
Kotlin version: 1.3.31
JDK version: 11.0.3
jUnit version: 5.1.1

V

**Comment by Jofairden** [comment]
I think I'm having the same issue atm. Trying to capture a 'JsonObject', which is marked `@Nullable`
I will actually get the error:
```
io.mockk.MockKException: 
Failed matching mocking signature for

left matchers: [slotCapture<JsonObject>()]
```
Which is just plain wrong, because the signature matches just fine. I tried captureNullable, but it only takes a mutable list. So now I also wonder why I can't capture this object, it makes me unable to complete my unit test.

Versions
MockK version: 1.9.3
OS: Win 7
Kotlin version: 1.3.30
JDK version: 1.8
jUnit version: 5.4.3
Kotlinx version: 1.2.1

Relevant code:
```kotlin
@Test
fun `a test`() {
	val fakeHandler: (JsonObject) -> Unit = {}
	val sendFormCallbackMockk: AsyncResult<HttpResponse<Buffer>> = mockk()

	val sendFormAsyncResultSlot = slot<Handler<AsyncResult<HttpResponse<Buffer>>>>()
	val handlerInSlot = slot<JsonObject>()
	every { req.sendForm(mapMockk, capture(sendFormAsyncResultSlot)) } just Runs

	// It will fail at this every block with the above error
	every { handlerMockk(capture(handlerInSlot)) } just Runs

	req.sendFormWithResultAsJson(mapMockk, fakeHandler)
	sendFormAsyncResultSlot.captured.handle(sendFormCallbackMockk)

	verify(exactly = 1) {
		req.sendForm(mapMockk, sendFormAsyncResultSlot.captured)
		// Need capture so I can verify here
		handlerMockk(handlerInSlot.captured)
	}
}
```

Is this just an oversight? The value of the captured slot can just stay null if it was null, right ?

**Comment by stengvac** [comment]
activity comment

**Comment by stengvac** [comment]
+1

**Comment by vistiyos** [comment]
Any update about this?

**Comment by oleksiyp** [comment]
List version going to capture one object per one call

**Comment by vistiyos** [comment]
Any estimation for the release 1.9.4?

**Comment by oleksiyp** [comment]
Ш

**Comment by oleksiyp** [comment]
I face problems with gradle builds, spent already few weekends, but not easy to fix

**Comment by cutiko** [comment]
Same problem here

**Comment by Raibaz** [comment]
Closing this as it looks like using the mutableList version of capturing slots solves it.

**Comment by mgj** [comment]
I still would very much like to be able to write something like:

`val arg1 = slot<String?>()`

slot is easier to use than mutableList

**Comment by andriygg** [comment]
it would be really cool to add support for capturing nullable values, like `slot<String?>()`

**Comment by Raibaz** [comment]
I see your point, @andriygg.

Currently you can work around it by using a mutableList rather than a slot, so I wouldn't consider adding support for nullable types in slots as a high priority feature request.

Anyways, a PR for it would definitely be cool :)

**Comment by stengvac** [comment]
Available in 1.13.7
